### PR TITLE
feat: add all mav2 contract addrs, abis

### PR DESCRIPTION
**Pull request by howydev**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on renaming several ABI exports for clarity and consistency, along with the addition of new ABI definitions for various modules in the smart contract codebase.

### Detailed summary
- Renamed `MAV2FactoryAbi` to `accountFactoryAbi`.
- Renamed `smaV2Abi` to `semiModularAccountBytecodeAbi`.
- Renamed `singleSignerValidationAbi` to `singleSignerValidationModuleAbi`.
- Updated import statements to reflect new names.
- Added `addresses` export with various module addresses.
- Introduced new ABIs for `paymasterGuardModule`, `timeRangeModule`, `webauthnValidationModule`, `nativeTokenLimitModule`, `allowlistModule`, `modularAccountAbi`, and `semiModularAccountStorageAbi`.

> The following files were skipped due to too many changes: `account-kit/smart-contracts/src/ma-v2/abis/semiModularAccountStorageAbi.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/smart-contracts/src/ma-v2/addresses.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// TODO: improve format
+
+export const addresses = {
+  allowlistModule:
+    "0xE46ca4a98c485caEE2Abb6ef5116292B8c78a868" as `0x${string}`,
```
Comment (linnall) [inline account-kit/smart-contracts/src/ma-v2/addresses.ts:5]: what is the purpose of `as 0x${string}`?

Comment (howydev) [reply account-kit/smart-contracts/src/ma-v2/addresses.ts:5]: without it, it'll default to type `string`

Comment (linnall) [reply account-kit/smart-contracts/src/ma-v2/addresses.ts:5]: oh I think I get it; it is enforcing the format for a hexadecimal string  👍

**Review by linnall — DISMISSED** [review]

File: account-kit/smart-contracts/src/ma-v2/abis/maV2.ts [hunk 2]
```diff
@@ -0,0 +1,1241 @@
+export const maV2Abi = [
```
Comment (jaypaik) [inline account-kit/smart-contracts/src/ma-v2/abis/maV2.ts:1]: Nit: Prefer to name this file after the contract it is from. If it is from `SemiModularAccountBytecode`, let's perhaps call it that? It seems we aren't consistent with naming between what we do for light account and msca though. Leaning towards sticking with the msca convention.

File: account-kit/smart-contracts/src/ma-v2/addresses.ts
Comment (jaypaik) [inline account-kit/smart-contracts/src/ma-v2/addresses.ts:1]: If we want to be consistent, it looks like addresses are under utils.ts for other accounts.

File: account-kit/smart-contracts/src/ma-v2/modules/single-signer-validation/abis/allowlistModule.ts
Comment (jaypaik) [inline account-kit/smart-contracts/src/ma-v2/modules/single-signer-validation/abis/allowlistModule.ts:1]: Not sure if these are all supposed to be under the single-signer-validation folder.

**Comment by howydev** [comment]
### Merge activity

* **Dec 13, 11:13 AM EST**: The merge label 'graphite-merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=alchemyplatform&repo=aa-sdk) once it meets the requirements.

**Comment by howydev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1207?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1207** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1207?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1207?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1203** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1203?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/alchemyplatform?ref=merge-queue-instructions-in-stack-comment&prId=1207).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Pull request merged**